### PR TITLE
Handling denomination precedence in scheduling.

### LINKED ISSUE #10: Schedule bug
@ralphbean ,

I think I stumbled across a bug in `statscache.schedule.Schedule`. I am not sure if it's a bug or feature :\

```
import datetime
import statscache.schedule

s = statscache.schedule.Schedule(minute=[0, 15, 30, 45])
now = datetime.datetime.strptime('2014-12-23 13:00', '%Y-%m-%d %H:%M')
s.next(now) # returns datetime.datetime(2014, 12, 23, 13, 0, 1)
s.next(s.next(now)) #returns datetime.datetime(2014, 12, 23, 13, 0, 2)

now = datetime.datetime.strptime('2014-12-23 13:01', '%Y-%m-%d %H:%M')
s.next(now) #returns datetime.datetime(2014, 12, 23, 13, 15, 1)
```

By initializing the above schedule, I want to run something every 15th minute of an hour, every hour. However, I see that, since I do not supply any `second` value, the schedule will run every second for the entire 15th minute. Is this a desired behaviour? Please correct me if I am wrong.

Let's say that the priority order of Schedule params is: `hour > minute > second`. If I initialize a schedule with a param, say `minute`, then we should accept default values for a higher param `hour` and not consider any values for a param with less priority: `second`. So, the above schedule will run every hour at the specified minutes.

Does it make sense?

### PR DESCRIPTION
**Pull request by ralphbean**
This should fix #10.

### CODE REVIEW

File: statscache/schedule.py [hunk 1]
```diff
@@ -37,6 +37,12 @@ class Hour(Denomination):
 
 class Schedule(object):
     def __init__(self, second=None, minute=None, hour=None):
+
+        # Let higher order denominations precede lesser ones
+        # https://github.com/fedora-infra/statscache/issues/10
+        second = second or (minute and [0])
+        minute = minute or (hour and [0])
```
Comment (dtgay) [inline statscache/schedule.py:44]: Fancy!

**Comment by dtgay** [comment]
LGTM :+1:

**Comment by ralphbean** [comment]
Thanks @oddshocks!

File: statscache/schedule.py (hunk 1 above)
Comment (pypingou) [inline statscache/schedule.py:44]: I wonder if I understand this correctly:

```
>>> None or (2 and [0])
[0]
>>> None or (1 and [0])
[0]
>>> None or ([0])
[0]
>>> None or (0 and [0])
0
```

Comment (ralphbean) [inline statscache/schedule.py:44]:
```
>>> None or (2 and [0])
[0]  # Correct,  there was no argument specified, but they did specify a higher-denomination argument, so use the [0] default here
>>> None or (1 and [0])
[0]  # Same as before.

>>> None or (0 and [0])
0  # This is invalid input and odd behavior, but a ``0`` will cause a traceback later on.
```

Comment (pypingou) [inline statscache/schedule.py:44]: No idea what kind of situation as possible or not, but just thought of another case:

```
>>> None or ([] and [0])
[]
```

**Pull request merged**